### ISSUE TITLE
Pressing p or SPACE doesn't pause the replay

### ISSUE THREAD

**Issue by shedatc** [OpenBW/openbw#23]
The `is_paused` variable in the `update()` method of `ui_functions` seems to be ignored.

I tried to add `bool is_paused` to the `replay_state` struct in the hope of being able to toggle it from `update()` and use it to from `replay_functions::next_frame()` to just skip frames but with no luck :/

In `ui.h`:
```c++
void update() {
    …
    if (e.sym == SDLK_SPACE || e.sym == SDLK_p) {
	is_paused = !is_paused;
	replay_st.is_paused = !replay_st.is_paused;
    …
}
```

In `replay.h`:
```c++
struct replay_state {
    …
    bool is_paused = false;
};

struct replay_functions: action_functions {
    …
    void next_frame() {
	if (replay_st.is_paused) return;
	…
    }
    …
```

The debugger confirms that two different replay_state structures seems to be allocated, i.e., `current_replay_state`'s address doesn't match `replay_st`'s one.

I'm only starting to try to grok this piece of code so any hint would be very much appreciated :D

**Comment by tscmoo** [comment]
So, just to be clear, this is when watching a replay with BWAPI?
Space/P doesn't pause the game when playing regular games either, so at least this is consistent.
The reason it doesn't work is because current_replay_state in ui_functions isn't the real replay_state, it's kind of only there because the struct it inherits from requires it (but it's not actually used, besides for pausing apparently).

**Comment by shedatc** [comment]
> So, just to be clear, this is when watching a replay with BWAPI?

Yes, running BWAPILauncher with a config file pointint to a replay (via the `map` parameter).

> The reason it doesn't work is because current_replay_state in ui_functions isn't the real replay_state, it's kind of only there because the struct it inherits from requires it (but it's not actually used, besides for pausing apparently).

So where is the _real replay_state_?

**Comment by shedatc** [comment]
Hmm, OK, I'll have a look. Thanks :)

**Comment by shedatc** [comment]
I proposed a patch to make pausing work again. Tell me what do you think :)
